pkg/controllers/cache/cache.go

```diff
@@ -154,17 +154,19 @@ func (jc *jobCache) Update(obj *v1alpha1.Job) error {
 		return fmt.Errorf("failed to find job <%v>", key)
 	}
 
-	var oldResourceversion, newResourceversion uint64
-	var err error
-	if oldResourceversion, err = strconv.ParseUint(job.Job.ResourceVersion, 10, 64); err != nil {
-		return fmt.Errorf("failed to parase job <%v> resource version <%s>", key, job.Job.ResourceVersion)
-	}
+	if job.Job != nil {
```

Comment:
When does job.Job=nil in update, i can see we set job.Job in Add callback

Reply:
as I mentioned in https://github.com/volcano-sh/volcano/issues/2417, when we delete vj, it triggers deleteJob() in informer of volcano-job-controller, deleteJob() calls cc.cache.Delete(job) which set **_jobInfo.Job = nil_** in **L182  pkg/controllers/cache/cache.go**